### DISCUSSION TITLE
Just an easy and simple dashboard for dummies

### DISCUSSION THREAD

**Discussion by netzkombyse** [solectrus/solectrus#1321] [General]
Hallo Georg,

was für ein fantastisches System du da entwickelt hast. Ich bin begeistert. 
Einerseits bin ich technisch eher unwahrscheinlich in der Lage, das bei mir nachzubauen (obwohl Synology NAS, Rasperry pi3, Senec-Kunde - aber ich bin technisch nicht versiert genug). 

Ich bin im Grund "nur" auf der Suche nach einem Dashboard bei dem ich auf einem ipad nur die linke Spalte (Herkunft & Verwendung mit Akkustand) sehe. Keine langszeitdoku, sondern nur "IST/Moment"-Zustand.

Meine Fragen daher: 

1. geht das? und wenn ja: 
2. Welche Teile aus den Repositories muss ich mir beibringen, damit ich das umsetzen kann?

Vielen Dank.
Henning

**Comment by netzkombyse** [comment]
Finally, ich kann mir fast selber die Antwort geben. Mit Hilfe eines guten Bekannten, der etwas mehr von Kommandozeilen-Architektur versteht, als ich, haben wir es geschafft. Ich musste auf meiner 718+ zwar mit SUDO-Befehlen Docker streicheln und musste einige /volume1/-Angaben ergänzen - aber wir haben es geschafft.

Bleibt dennoch die Frage: Ich brauche den Graphen und die Datenbank eigentlich nicht. Wo kann ich das Frontend "reduzieren"? Wenn die NAS ausgeschaltet wird, erledigt sich der DB-Zugriff ja eh - aber auch so, würde ich da gerne grundsätzlich drauf verzichten.

Vielen Dank für eine Handreichung.

**Comment by ledermann** [comment]
Hallo Henning, zuerst mal vielen Dank für die Blumen - freut mich, dass Solectrus gefällt und auch die Installation geklappt hat :)

Bezüglich der von Dir nachgefragten Reduzierung der Systemkomponenten: Möglichkeiten für einen Wegfall der Zeitreihen-Datenbank sehe ich eher nicht. Es ist ja so, dass das Dashboard nicht direkt auf die SENEC-Hardware zugreift, sondern der SENEC-Collector (als eigenständiger Docker-Container) fragt die SENEC-Hardware im 5-Sekunden-Takt ab und schreibt die Daten in die InfluxDB, von wo sie dann bei Bedarf vom Dashboard abgerufen werden:

```mermaid
flowchart LR
   SENEC-Hardware -- SENEC-Collector --> InfluxDB --> Dashboard
```

Für mich war die dauerhafte Speicherung wichtig, weil nur dadurch auch Zeiträume dargestellt und Kosten/Gewinn-Aspekte untersucht werden können. Dadurch muss die Synology natürlich rund um die Uhr laufen, was bei mir aber aus anderen Gründen sowieso der Fall ist.

Grundsätzlich wäre es zwar denkbar, ein Dashboard zu entwickeln, dass die Live-Messwerte direkt von der SENEC abholt und ohne Zwischenspeicherung auskommt, aber das wäre ein ganz anderes Konzept. Das würde auch ganz ohne Server auskommen, also nur mit JavaScript, ohne Ruby-Backend. Eine solch vollkommen andere App habe ich nicht in Planung.

BTW: Die derzeitig auf https://solectrus.ledermann.dev zu sehende Strombilanz-Darstellung mit Herkunft und Verwendung ist noch work-in-progress (siehe #948) und noch nicht im offiziellen Docker-Image enthalten, das dauert vermutlich noch eine Weile.

**Comment by spel-ux** [comment]
Hallo,

ein grandioses Dashboard!

Da auch bald alle erforderlichen Komponenten erhalte, möchte auch ich Solectrus bei mir installieren.

Wenn ich es richtig verstehe, ist beim In-Hous-Docker Image alles dabei? Lediglich die Config muss angepasst werden?
Bei mir läuft die Hausautomationssoftware FHEM auf einem Atom x5-Z8350, das sollte reichen. Ich würde gerne auch per FHEM Daten auslesen (https://fhem.de/). Hier wäre dann auf die Datenbank zuzugreifen?

Vielen Dank!

**Comment by ledermann** [comment]
@spel-ux Ja, es ist alles dabei, was man benötigt. Die Gesamtinstallation verteilt sich auf verschiedene Docker-Images, wobei das Dashboard die Hauptanwendung darstellt.

FHEM kenne ich selbst nicht. Aber es ist durchaus denkbar, dass andere Anwendungen auf die InfluxDB zugreift, die von Solectrus verwaltet wird.

**Comment by CapoDecina89** [comment]
@ledermann Hallo Georg,

erstmal vielen Dank für das tolle Dashboard.
Ich habe alles nach deiner Anleitung für das Hosting auf verteilten Systemen eingerichtet. Allerdings bin ich mir nicht sicher wie ich den Server konfigurieren muss um per SSL/TLS auf die App zuzugreifen? Könntest du das kurz erklären?

Beste Grüße!

**Comment by ledermann** [comment]
@CapoDecina89 Die Einrichtung von SSL/TLS für den Zugriff per https ist **nicht** Teil meiner Anleitung.

Wenn man das haben möchte, wird es etwas komplizierter. Ich verwende z.B. [Traefik](https://github.com/traefik/traefik) als vorgeschalteten Proxy, der das Domain-Handling sowie automatisch Zertifikate per Let’s Encrypt erstellt. Jeder eingehende (https-)Request wird dann an den passenden Service (wie z.B. Solectrus) weitergeleitet. 

Für Traefik gibt es ebenfalls Docker images. Die Installation lässt sich aber nicht kurz erklären, da möchte ich auf die vielen Tutorials und Anleitungen verweisen, die man im Netz finden kann.

**Comment by GerhardPg** [comment]
Hallo Georg,
das Dashboard läuft jetzt seit einer guten Woche auf meinem NAS und ich bin begeistert, weil es genau dem entspricht was ich mir vorgestellt und gesucht habe. Vielen Dank für die Veröffentlichung. 
Prinzipiell gefällt mir auch der Forecast für den aktuellen Tag und den Folgetag. Allerdings ist dieser bei mir immer zu hoch, was an der Verteilung der PV-Module auf beiden Seiten des Satteldachs und der Garage liegt. In der Konfigurationsdatei kann ich ja nur ein Paket mit gleicher Ausrichtung eingeben, oder habe ich da was übersehen?

**Reply by ledermann** [reply]
Besten Dank, freue mich immer über solche Rückmeldungen :)

Zum Forecast: Unterstützung für mehrere Dächer ist bereits implementiert, aber noch nicht im `stable`-Image angekommen. Du kannst aber das `develop`-Image verwenden und testen, genaueres hier:
https://github.com/solectrus/forecast-collector/pull/207

Bei weiteren Fragen dazu besser ein Issue hier aufmachen:
https://github.com/solectrus/forecast-collector/issues

**Comment by GerhardPg** [comment]
Hallo Georg,

danke für die Info. Bin jetzt auch endlich dazu gekommen, das .env-File entsprechend anzupassen, auf dem NAS zu ersetzen und einen Reboot durchzuführen. Reicht das, oder muss ich mehr tun?

Meine Programmierkenntnisse sind leider eher rudimentär.

Der Forecast für morgen hat in etwa die „Figur“ des erwarteten Produktionsverlaufs und auch das Maximum kurz nach Mittag entspricht dem, was in den letzten Tagen so produziert wurde.

FORECAST_CONFIGURATIONS=4

FORECAST_0_DECLINATION=40

FORECAST_0_AZIMUTH=-53.6

FORECAST_0_KWP=5.625

FORECAST_1_DECLINATION=40

FORECAST_1_AZIMUTH=126.4

FORECAST_1_KWP=4.125

FORECAST_2_DECLINATION=10

FORECAST_2_AZIMUTH=36.4

FORECAST_2_KWP=1.125

FORECAST_3_DECLINATION=10

FORECAST_3_AZIMUTH=-143.6

FORECAST_3_KWP=1.125

 

Viele Grüße

Gerhard

 

Von: Georg Ledermann ***@***.***> 
Gesendet: Montag, 25. September 2023 10:32
An: solectrus/solectrus ***@***.***>
Cc: GerhardPg ***@***.***>; Comment ***@***.***>
Betreff: Re: [solectrus/solectrus] Just an easy and simple dashboard for dummies (Discussion #1321)

 

Besten Dank, freue mich immer über solche Rückmeldungen :)

Zum Forecast: Unterstützung für mehrere Dächer ist bereits implementiert, aber noch nicht im stable-Image angekommen. Du kannst aber das develop-Image verwenden und testen, genaueres hier:
solectrus/forecast-collector#207 <https://github.com/solectrus/forecast-collector/pull/207> 

Bei weiteren Fragen dazu besser ein Issue hier aufmachen:
https://github.com/solectrus/forecast-collector/issues

**Reply by ledermann** [reply]
Wenn du vier Dachflächen hast und so präzise deren Ausrichtung angeben kannst, dann sieht alles korrekt aus :) Reboot wäre übrigens nicht nötig gewesen, Neustart des Forecast-Collector-Containers genügt.

Bei weiteren Fragen dazu bitte nicht diese schon sehr unübersichtliche Diskussion fortsetzen, sondern ein neues Issue hier eröffnen: https://github.com/solectrus/forecast-collector/issues
